api/v1alpha1/types.go

```diff
@@ -0,0 +1,116 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	runtime "k8s.io/apimachinery/pkg/runtime"
+)
+
+type ProvisionerID string
+
+// ProvisionerClassSpec defines the desired state of ProvisionerClass
+type ProvisionerClassSpec struct {
+	Provisioner ProvisionerID `json:"provisioner"`
+
+	// +optional
+	Parameters map[string]string `json:"parameters,omitempty"`
+}
+
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:storageversion
+// +kubebuilder:resource:categories=rukpak,scope=Cluster
+// +kubebuilder:subresource:status
+
+// ProvisionerClass is the Schema for the provisionerclasses API
+type ProvisionerClass struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec ProvisionerClassSpec `json:"spec"`
+}
+
+// +kubebuilder:object:root=true
+
+// ProvisionerClassList contains a list of ProvisionerClass
+type ProvisionerClassList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []ProvisionerClass `json:"items"`
+}
+
+// Source locates content to stage.
+type Source string
+
+// BundleSpec defines the desired state of Bundle
+type BundleSpec struct {
+	// Class specifies the name of the provisioner that should manage the Bundle.
+	// +optional
+	Class string `json:"class,omitempty"`
+
+	// Sources locates all remote content to be staged by the Bundle.
```

Comment:
This could also be local content but this description is fine with me for now until we expand the provisioner's scope.

Reply:
Yeah, I was thinking that we could use RawExtension and let provisioners ducktype this field too.

Let's stick with this and see where it falls down (we are in `v1alpha1` after all).